File: leetcode_796/RotateString.java
package leetcode_796;

class RotateStirng{
    public boolean rotateString(String s, String goal) {
        if (s.length() != goal.length()) return false;
        
        String contatenate = s + s;
        
        return contatenate.contains(goal);
    }

    public static void main(String[] args) {
        RotateStirng rs = new RotateStirng();
        String s = "abcde";
        String goal = "cdeab";
        System.out.println(rs.rotateString(s, goal));
    }
}

File: leetcode_796/RotateString.py
class RotateStirng:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False
        contatenate = s + s
        return goal in contatenate

def main():
    s = "abcde"
    goal = "cdeab"
    solution = RotateStirng()
    print(solution.rotateString(s, goal))

if __name__ == "__main__":
    main()